drivers/cc2420/cc2420.c

```diff
@@ -197,8 +197,8 @@ int cc2420_rx(cc2420_t *dev, uint8_t *buf, size_t max_len, void *info)
         DEBUG("cc2420: recv: reading %i byte of the packet\n", (int)len);
         cc2420_fifo_read(dev, buf, len);
 
-        uint8_t rssi;
-        cc2420_fifo_read(dev, &rssi, 1);
+        int8_t rssi;
```

Comment:
why change this to signed, and then cast it to unsigned below? Seems a bit odd to me.

Reply:
Probably a leftover from the first implementation that used `uint8_t`s. Will fix.

Reply:
Ignore my previous comment. @smlng Assuming you're referring to the cast in the `cc2420_fifo_read()` function, that function requires a `uint8_t*` to write the result of the read to. The value from the cc2420 register should be interpreted as "signed 2’s complement".

Reply:
ah okay, understood